File: ShadowFrameworkLite20/src/shadow/image/bitmaps/data/SFVoronoiBitmapFunctionData.java
package shadow.image.bitmaps.data;

import java.util.ArrayList;

import shadow.geometry.SFValuesList;
import shadow.image.bitmaps.SFBasicVoronoiDistances;
import shadow.image.bitmaps.SFBasicVoronoiModels;
import shadow.image.bitmaps.SFBitmapFunction;
import shadow.image.bitmaps.SFVoronoiBitmapFunction;
import shadow.math.SFValuenf;
import shadow.math.SFVertex2f;
import shadow.renderer.data.SFDataAsset;
import shadow.renderer.data.SFLibraryReference;
import shadow.system.data.SFNamedParametersObject;
import shadow.system.data.objects.SFEnumObject;

public class SFVoronoiBitmapFunctionData extends SFDataAsset<SFBitmapFunction>{

	protected SFLibraryReference<SFValuesList<SFValuenf>> vertices = 
			new SFLibraryReference<SFValuesList<SFValuenf>>();

	private static String[] modelsNames=SFBasicVoronoiModels.names();
	
	private static SFBasicVoronoiModels[] models=SFBasicVoronoiModels.values();
	
	private SFEnumObject<SFBasicVoronoiModels> model=new SFEnumObject<SFBasicVoronoiModels>(models,modelsNames);
	
	private static String[] distancesNames=SFBasicVoronoiDistances.names();
	
	private static SFBasicVoronoiDistances[] distances=SFBasicVoronoiDistances.values();
	
	private SFEnumObject<SFBasicVoronoiDistances> distance=new SFEnumObject<SFBasicVoronoiDistances>(distances,distancesNames);
	
	public SFVoronoiBitmapFunctionData() {
		super();
		SFNamedParametersObject parameters=new SFNamedParametersObject();
		parameters.addObject("vertices", vertices);
		parameters.addObject("model", model);
		parameters.addObject("distance", distance);
		setData(parameters);
	}
	
	@Override
	protected SFBitmapFunction buildResource() {
		SFVoronoiBitmapFunction function=new SFVoronoiBitmapFunction();
		function.setDistance(distance.getElement());
		function.setModel(model.getElement());
		SFValuesList<SFValuenf> vertices=this.vertices.getDataset().getResource();
		ArrayList<SFVertex2f> centers=new ArrayList<SFVertex2f>();
		for (int i = 0; i < vertices.getSize(); i++) {
			SFVertex2f vertex=new SFVertex2f(0,0);
			vertices.getValue(i, vertex);
			centers.add(vertex);
		}
		function.setCenters(centers);
		return function;
	}
	
}

File: trunk/ShadowFrameworkJS/src/shadow/image/bitmaps/data/SFVoronoiBitmapFunctionData.js

var SFVoronoiBitmapFunctionData_models=SFBasicVoronoiModels_values();
		
var SFVoronoiBitmapFunctionData_distances=SFBasicVoronoiDistances_values();

function SFVoronoiBitmapFunctionData(){
	
	this.vertices =new SFLibraryReference();
	this.model=new SFEnumObject(SFVoronoiBitmapFunctionData_models);
	this.distance=new SFEnumObject(SFVoronoiBitmapFunctionData_distances);


	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.vertices);
	parameters.addObject(this.mdoel);
	parameters.addObject(this.distance);
	this.setData(parameters);
}

inherit(SFVoronoiBitmapFunctionData,SFDataAsset);


SFVoronoiBitmapFunctionData.prototype["buildResource"]=function(){
	var func=new SFVoronoiBitmapFunction();
	func.setDistance(this.distance.getElement());
	func.setModel(this.model.getElement());
	var vertices=this.vertices.getDataset().getResource();
	var centers=new Array();
	for (var i = 0; i < vertices.getSize(); i++) {
		var vertex=new SFVertex2f(0,0);
		vertices.getValue(i, vertex);
		centers.push(vertex);
	}
	func.setCenters(centers);
	return func;
};


SFVoronoiBitmapFunctionData.prototype["generateNewDatasetInstance"]=function(){
	return new SFVoronoiBitmapFunctionData();
};


	
	
